### PR TITLE
Minor REST API improvements

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | Slack request from QA
| Versions      | 4.5+
| Edition       | N/A

- Add/Fix Route `orders/order/cancel/{id}`.
- Code change of “No content” response from 200-ok to 204.
- Example simplification in `/product/catalog/products/{productTypeIdentifier}`

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Redirects cover removed/moved pages
- [x] Code samples are working
- [x] PHP code samples have been fixed with PHP CS fixer
- [x] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

**Review by juskora — APPROVED** [review]

File: docs/api/rest_api/rest_api_reference/input/examples/product/catalog/product/POST/ProductCreate.json.example [hunk 1]
```diff
@@ -32,10 +32,8 @@
             "mainLanguageCode": "eng-GB"
         },
         "code": "ggg",
-        "attributes": [
-            {
+        "attributes": {
                 "attr1": "fffffff"
             }
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/examples/product/catalog/product/POST/ProductCreate.json.example:37]: Fix indent:
```suggestion
            "attr1": "fffffff"
        }
```

File: docs/api/rest_api/rest_api_reference/input/ez-orders.raml [hunk 2]
```diff
@@ -76,69 +76,70 @@ displayName: Orders
...
-                401:
-                    description: Error - the user is not authorized to delete this order.
-                404:
-                    description: Error - the order with the given ID does not exist.
-                406:
-                    description: Error - the order cannot be cancelled due to not-allowed workflow transition
+    /cancel:
+        /{id}:
+            get:
+                displayName: Load order by ID
```
Comment (adriendupuis) [inline docs/api/rest_api/rest_api_reference/input/ez-orders.raml:82]: It doesn't make much sense that `/orders/order/cancel/{id}` loads an order.

Comment (adriendupuis) [reply docs/api/rest_api/rest_api_reference/input/ez-orders.raml:82]: [`/orders/order/{id}`](https://github.com/ibexa/order-management/blob/main/src/bundle/Resources/config/routing_rest.yaml#L9) should be kept.

[`/orders/order/cancel/{id}`](https://github.com/ibexa/order-management/blob/main/src/bundle/Resources/config/routing_rest.yaml#L39) is a whole new route.

Comment (adriendupuis) [reply docs/api/rest_api/rest_api_reference/input/ez-orders.raml:82]: It should looks like
```yaml
displayName: Orders

/orders:
    post:
        …
    get:
        …
/order:
    /{identifier}:
        get:
            …

    /{id}:
        get:
            …
        patch:
            …
        post:
            …
    /cancel/{id}:
        post:
            displayName: Cancel order
            …
```

Comment (adriendupuis) [reply docs/api/rest_api/rest_api_reference/input/ez-orders.raml:82]: In fact, it's just the insertion of `    /cancel/{id}:` at line 133

**Review by adriendupuis — CHANGES_REQUESTED** [review]
The insertion of `    /cancel/{id}` at line 133 without any indentation change should keep every existing routes and just fix the `Cancel order` one.

**Review by adriendupuis — APPROVED** [review]
Everything looks good
* https://ez-systems-developer-documentation--2131.com.readthedocs.build/en/2131/api/rest_api/rest_api_reference/rest_api_reference.html#product-catalog-create-product
* https://ez-systems-developer-documentation--2131.com.readthedocs.build/en/2131/api/rest_api/rest_api_reference/rest_api_reference.html#orders-cancel-order

**Pull request merged**